misc/metrics/src/swarm.rs

```diff
@@ -20,15 +20,16 @@
 
 use crate::protocol_stack;
 use prometheus_client::encoding::text::Encode;
-use prometheus_client::metrics::counter::Counter;
-use prometheus_client::metrics::family::Family;
+use prometheus_client::metrics;
+use metrics::{counter::Counter,family::Family,histogram::Histogram};
 use prometheus_client::registry::Registry;
 
 pub struct Metrics {
     connections_incoming: Family<AddressLabels, Counter>,
     connections_incoming_error: Family<IncomingConnectionErrorLabels, Counter>,
 
     connections_established: Family<ConnectionEstablishedLabels, Counter>,
+    connection_duration: Family<AddressLabels,Histogram>,
```

Comment:
I suggest also tracking whether this was an inbound or outbound connection.

Reply:
Updated description to show what it looks like now.